### ISSUE TITLE
Token not shown in Safari ("Error: Stored state not found")

### ISSUE THREAD

**Issue by cjsinnbeck** [luukhaijes/kinde-angular#18]
**Describe the bug**
When using the login() method on the Angular service, the token is not available in Safari after I am redirected from Kinde back to my Angular app. Instead I get "Error: Stored state not found" from AuthCodeWithPKCE.js (line 180). It works perfectly well in Chrome.

**To Reproduce**
For a simple test, include the service in a component constructor:

```
constructor(
    public kindeAuthService: KindeAngularService
) {}
```

Make a simple login button in the component's view and display the token right beneath it:

```
<div id="logged-out-buttons">
  <button id="login" type="button" (click)="kindeAuthService.login()">Login</button>
  <div>
    {{ kindeAuthService.accessToken$ | async }}
  </div>
</div>
```

In Safari, the token is null, and the console displays the error mentioned above. In Chrome, the access token is displayed.

**Expected behavior**
I expected the same behavior as in Chrome, where the token is accessible.

**Desktop (please complete the following information):**
 - OS: MacOS 12.3
 - Browser: Safari 15.4
 - "kinde-angular": "^1.3.0"

**Comment by luukhaijes** [comment]
Odd, I will have a look at this.

**Comment by luukhaijes** [comment]
Noticed that the cookie policy of safari is more restricted compared to other browsers. This wont happen in production environment if you use the same domain e.g.: app.yourdomain.com and auth.yourdomain.com. Nevertheless, I will search for solution to make localhost work again

**Comment by cjsinnbeck** [comment]
@luukhaijes Thanks a lot for investigating this. I was aware of the cookie issues in Safari from some previous problems I had with Auth0 (which also relies on third party cookies), so I am almost sure that I had allowed third party cookies in my Safari settings when I did the Kinde test. I will confirm when I am at my computer and get back to you here.

**Comment by luukhaijes** [comment]
No problem, please let me know so I can continue my investigation. :)

**Comment by luukhaijes** [comment]
@cjsinnbeck I tried it myself with website tracking prevention turned off, same problem occurs. Thanks for raising this issue, thinking of a fix right now.

**Comment by cjsinnbeck** [comment]
@luukhaijes Thanks for confirming. It seems to me that the error does not come from your code but rather from Kinde's own npm package (Kinde TypeScript SDK), and that it has to do with some SessionStorage management? The error I get seems to come from here:

https://github.com/kinde-oss/kinde-typescript-sdk/blob/a9f596bd136ee0412d3eb3b4e3c8e1aca1473421/lib/sdk/oauth2-flows/AuthCodeWithPKCE.ts#L119

So, maybe this isn't even a cookie issue?

**Comment by luukhaijes** [comment]
@cjsinnbeck Yeah, that happens when there is no storedState aka the cookie. It uses the sessionManager where I created a custom implementation. I noticed no cookies are being set, currently I'm changing the implementation to use the refreshTokens function, needs some work.

**Comment by cjsinnbeck** [comment]
@luukhaijes Ah, got it. (I am having a very similar problem with Auth0 + Safari, where I need to change a project to use RefreshTokens + LocalStorage instead of cookies to make it work in Safari). 

I also see some StackOverflow answers saying that Safari/Webkit will not set secure cookies for unrecognized TLDs, and that "localhost" is such an unrecognized TLD. This is a separate cookie issue unrelated to the "third party coookie" issue. So, maybe that is part of the problem?

Anyway, thanks a lot for looking into this.

**Comment by luukhaijes** [comment]
Working on this, I'm reworking the package a bit so I hope to have this fixed soon :)
